### DISCUSSION TITLE
Is music_none.cpp necessary anymore?

### DISCUSSION THREAD

**Discussion by alsed** [raceintospace/raceintospace#876] [Q&A] [answered]
One of the source files that deals with the music is music_none.cpp

Inside, it declares:
* This file handles music ignorance.
* Its purpose is to compile, when no music would be provided at all. */

I assume this file was created when the music was sent to a non-free repository, so the game was by default compiling without it.
In short,  is a copy of music_vorbis.cpp but with an empty function to load music files.
`void music_load(enum music_track track)
{
}`

As this discussion implies:
https://github.com/raceintospace/raceintospace/discussions/708
the problem to compile without the music files seem to have been resolved.
Also, I've compiled the game without this file and also without the music, and I've had no problems since the code knows how to handle situations when it can't find or load the music files.

So the question is, is this source file needed anymore? Can it be safely removed?

**Comment by peyre** [comment]
Alsed, with the contributions you've been making - especially fixing #209 - I'd like to add you to the credits if you're ok with that.  If you'd like to be, I'll need your real name (or at least what you want to show there).  You can email me (lordpeyre@yahoo.com) if you don't want it advertised here.

**Reply by alsed** [reply]
Thank you @peyre!, though that fix was primarily made by @kkosciusz , I just typed the code.
I hope to contribute much more to the Race Into Space, I dream with translate it to other languages.
I'll send you an email.

**Comment by alsed** [comment]
After pull #866 music_none.cpp is no longer compiled in cmake, so this source file can be safely removed.

**Reply by kkosciusz** [reply, marked as answer]
I think music none may be useful for test environments, but then it can be brought back from the git.

**Reply by alsed** [reply]
Last time I tried to delete the entire test folder, lol.
I know it is important for cmake, I hope I can have time to learn how it works.
